### PR TITLE
Bug/31 add type name to column adapter notes

### LINKED ISSUE #31: Add type name to `ColumnAdapterNotes`
## Situation

The "type name" attribute of the `ColumnAdapterNotes` was removed in one of the recent refactorings. Since this is still needed in some of the dialects, we have to restore it.

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #31.

### CODE REVIEW

File: src/main/java/com/exasol/adapter/adapternotes/ColumnAdapterNotes.java [hunk 1]
```diff
@@ -26,20 +25,107 @@ public int getJdbcDataType() {
         return this.jdbcDataType;
     }
 
+    /**
+     * Get the type name
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/adapter/adapternotes/ColumnAdapterNotes.java:29]:
```suggestion
     * Get the type name.
```

File: src/main/java/com/exasol/adapter/adapternotes/ColumnAdapterNotesJsonConverter.java [hunk 2]
```diff
@@ -10,6 +10,8 @@
  */
 public final class ColumnAdapterNotesJsonConverter {
     private static final ColumnAdapterNotesJsonConverter COLUMN_ADAPTER_NOTES_JSON_CONVERTER = new ColumnAdapterNotesJsonConverter();
+    static final String JDBC_DATA_TYPE = "jdbcDataType";
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/adapter/adapternotes/ColumnAdapterNotesJsonConverter.java:13]: Why is one variable public and another package-private? Do you need only one of them outside the package?

Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/adapternotes/ColumnAdapterNotesJsonConverter.java:13]: Maybe it's better to use a getter if you need a public variable?

Comment (redcatbear) [reply src/main/java/com/exasol/adapter/adapternotes/ColumnAdapterNotesJsonConverter.java:13]: The second one is used in a whitebox unit test.

**Pull request merged**